### DISCUSSION TITLE
Turbo instead of turbolinks

### DISCUSSION THREAD

**Discussion by Vagab** [luckyframework/lucky#1600] [Feature requests & Ideas]
Hey everyone! I just started using Lucky, coming from long years of ruby development. And I noticed it uses turbolinks. Rails has shown new [Turbo](https://github.com/hotwired/turbo-rails) which seems better from what I've heard and from my own experience. Why not include it as a default? If there is a plan to do so but not enough developer power, I can try to help out.

**Comment by jwoertink** [comment]
Hey! Welcome to Lucky. We sort of started a discussion on this https://github.com/luckyframework/lucky_cli/issues/629 Though, it was more about trying hotwire. I haven't used Turbo before, but if enough people are using it in their Lucky apps, I'm open to using it. I just really don't know what all it entails, and how well it plays with Lucky.

Maybe @stephendolan or @snadon can chime in on this?

**Reply by Vagab** [reply]
Well, afaik Turbo is a part of Hotwire and is just a successor of turbolinks. But I meant Hotwire as a whole, definitely.

**Reply by stephendolan** [reply]
Yeah, I think it'd be useful to enumerate the ways in which Turbo is better than Turbolinks for what Lucky is using it for. The Hotwire stack would be a massive improvement in what's possible for dynamic pages, but would also require quite a lot of documentation and changes to support things like Turbo Frames.

A pull request with a switch to Turbo would probably be the best place to have a discussion about its merits since we could pull down the app and play with it.

**Reply by jwoertink** [reply]
I like that idea. It would be nice to see a simple bullet list on why Turbo is better, and what it provides us. Plus it gives us a chance to really see what is involved with making that change. Also, having the escape hatch to back out if it's not wanted.

**Comment by skojin** [comment]
I have a simple demo with lucky + turbo. 
form saving with validation, update with redirect vs turbo-frame vs turbo-stream

https://lucky-turbo-demo.herokuapp.com

https://github.com/skojin/lucky_turbo_demo/commit/27838f0defb36b3824c0e7b6cc0811ab9e60f8b7

**Comment by patleb** [comment]
Would it be possible to keep backward compatibility with Turbolinks?

**Reply by jwoertink** [reply]
Maybe for the first release after moving over, but we would most likely deprecate and remove fairly soon since Turbolinks itself is [deprecated and archived](https://github.com/turbolinks/turbolinks-classic).

**Reply by patleb** [reply]
Alright. Although, Turbolinks 5 is still supported by Rails for Tubo-iOS and Turbo-Android adapters. Keeping backward compatibility might forces you to keep unwanted code around, but it's very minimal. In my case, the main reason for still using an old version of Turbolinks is the need to support old android platforms (4.1 to 4.4).

**Reply by jwoertink** [reply]
Oh? Is this a Lucky app you can share? We can add it to the list https://luckyframework.org/awesome

**Reply by patleb** [reply]
No, sorry, but thanks for the link.
